x/vesting/keeper/hooks.go

```diff
@@ -0,0 +1,207 @@
+// the module keeper
+func (h Hooks) AfterProposalFailedMinDeposit(ctx sdk.Context, proposalID uint64) {
+	h.k.AfterProposalFailedMinDeposit(ctx, proposalID)
+}
+
+// AfterProposalFailedMinDeposit is called after a governance clawback proposal fails due to
+// not meeting the minimum deposit.
+func (k Keeper) AfterProposalFailedMinDeposit(ctx sdk.Context, proposalID uint64) {
+	proposal, isClawbackProposal, err := k.getClawbackProposal(ctx, proposalID)
+	if err != nil {
+		k.Logger(ctx).Error("failed to check proposal",
+			"proposalID", proposalID,
+			"hook", "AfterProposalFailedMinDeposit",
+			"error", err,
+		)
+		return
+	}
+	if !isClawbackProposal {
+		// no-op when proposal is not a clawback proposal
+		return
+	}
+
+	vesting, funder, err := k.getAddressAndFunder(ctx, proposal)
+	if err != nil {
+		k.Logger(ctx).Error(
+			"failed to get clawback vesting account",
+			"proposalID", proposalID,
+			"hook", "AfterProposalFailedMinDeposit",
+			"error", err,
+		)
+		return
+	}
+
+	k.DeleteActiveClawbackProposal(ctx, vesting, funder)
+}
+
+// AfterProposalVotingPeriodEnded is a wrapper for calling the Gov AfterProposalVotingPeriodEnded hook on
+// the module keeper
+func (h Hooks) AfterProposalVotingPeriodEnded(ctx sdk.Context, proposalID uint64) {
+	h.k.AfterProposalVotingPeriodEnded(ctx, proposalID)
+}
+
+// AfterProposalVotingPeriodEnded is called after the voting period of a governance clawback proposal
+// has ended.
+func (k Keeper) AfterProposalVotingPeriodEnded(ctx sdk.Context, proposalID uint64) {
+	proposal, isClawbackProposal, err := k.getClawbackProposal(ctx, proposalID)
+	if err != nil {
+		k.Logger(ctx).Error("failed to check proposal",
+			"proposalID", proposalID,
+			"hook", "AfterProposalVotingPeriodEnded",
+			"error", err,
+		)
+		return
+	}
+	if !isClawbackProposal {
+		// no-op when proposal is not a clawback proposal
+		return
+	}
+
+	vesting, funder, err := k.getAddressAndFunder(ctx, proposal)
+	if err != nil {
+		k.Logger(ctx).Error(
+			"failed to get clawback vesting account",
+			"proposalID", proposalID,
+			"hook", "AfterProposalVotingPeriodEnded",
+			"error", err,
+		)
+		return
+	}
+
+	k.DeleteActiveClawbackProposal(ctx, vesting, funder)
+}
+
+// getClawbackProposal checks if the proposal with the given ID is a governance
+// clawback proposal.
+func (k Keeper) getClawbackProposal(ctx sdk.Context, proposalID uint64) (vestingtypes.ClawbackProposal, bool, error) {
+	proposal, found := k.govKeeper.GetProposal(ctx, proposalID)
+	if !found {
+		return vestingtypes.ClawbackProposal{}, false, errors.New("proposal not found")
+	}
+
+	// TODO: do we have to check here or is this hook only called for clawback governance proposals?
+	msgs, err := proposal.GetMsgs()
+	if err != nil {
+		return vestingtypes.ClawbackProposal{}, false, err
+	}
+	if len(msgs) == 0 {
+		return vestingtypes.ClawbackProposal{}, false, errors.New("proposal has no messages")
+	}
+
+	msgContent, ok := msgs[0].(*govv1.MsgExecLegacyContent)
```

Comment:
shouldn't this iterate? could it be the case that there are multiple proposal msgs?